ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -110,6 +110,23 @@ def _process_visual_observation(
     return np.array(batched_visual, dtype=np.float32)
 
 
+def _raise_on_nan_and_inf(data: np.array, source: str) -> np.array:
+    # Check for NaNs or infs in the observations
+    # If there's a NaN in the observations, the np.mean() result will be NaN
+    # If there's an Inf (either sign) then the result will be Inf
+    # See https://stackoverflow.com/questions/6736590/fast-check-for-nan-in-numpy for background
+    # Note that a very large values (larger than sqrt(float_max)) will result in an Inf value here
+    # This is OK though, worst case it results in an unnecessary (but harmless) nan_to_num call.
+    d = np.mean(data)
```

Comment:
If data is empty (e.g. no vector obs but visual obs) then the mean is a NaN, and it will trigger the exception

Reply:
I don't _think_ that can happen (if you only have visual obs, we shouldn't send an empty array for the "vector observations") but maybe worth checking for.

Reply:
This is currently happening in the tests - not sure if it's limited to the demonstration loader or if it can happen elsewhere too. Might also be an incompatibility between the old demos and the new BatchedStepResult based API.

Reply:
so is the consensus here to check for empty, and early out if it is?

Reply:
That's my vote. @chriselion?

Reply:
Sounds reasonable.

Reply:
alright, thanks.